web-registry/src/lib/ga.ts

```diff
@@ -54,21 +56,32 @@ function isCookiesEnabled(): boolean {
   );
 }
 
-export function init(): void {
-  if (process.env.NODE_ENV === 'production') {
-    if (
-      Cookies.get(
-        COOKIE_GATSBY_PLUGIN_GOOGLE_ANALYTICS_GDPR_COOKIES_ENABLED,
-      ) === undefined
-    ) {
-      Cookies.set(
-        COOKIE_GATSBY_PLUGIN_GOOGLE_ANALYTICS_GDPR_COOKIES_ENABLED,
-        'false',
-      );
+export function useGoogleAnalyticsInit(): void {
+  const location = useLocation();
+  const [initialized, setInitialized] = useState(false);
+
+  useEffect(() => {
+    if (process.env.NODE_ENV === 'production') {
```

Comment:
one thing i need help with is figuring out how to test this. i think at present we only have google analytics running for the `master` branch (NODE_ENV=production). it's a small set of changes so far, but i would still feel better knowing it's been tested. i suppose i will need to create a temporary tag for my localhost or something... if anyone knows how i can test or have advice, please LMK.

Reply:
Actually, all netlify builds should be run with NODE_ENV production. However, it's possible that only the events on the production domain are saved. I think it's possible to keep all events by defining environments in GA: https://support.google.com/tagmanager/answer/6311518?hl=en

Reply:
It turns out that we're sending data for even the deploy previews to our google analytics setup. We'll probably want to change that before going live, i.e. we probably want to have something like prod has it's own analytics and perhaps redwood, hambach, and deploy previews all have their own separated analytics. Currently it just all getes sent to the same analytics bucket. Thanks for point that out @flagrede .

I was able to test this by going to the deploy preview and I could see that there was real-time analytics coming through while I was clicking around (even on the "trade" and "activity" tabs, which i don't think were working previously).

@erikalogie i think you could test this by going here:

- https://deploy-preview-1120--regen-registry.netlify.app/
- https://analytics.google.com/analytics/web/#/realtime/rt-overview/a119338253w247735863p229931745/

the second link should bring you to the real-time view in GA for the "Regen Network (new site)" analytics, and when you're clicking around you should be able to see your activity where you hadn't previously.

Reply:
i created a new issue for making the current google analytics setup only receive production data: https://github.com/regen-network/regen-registry/issues/1133